pkg/virt-controller/watch/migration.go

```diff
@@ -332,7 +335,7 @@ func (c *MigrationController) execute(key string) error {
 		return err
 	}
 
-	needsSync := c.podExpectations.SatisfiedExpectations(key) && vmiExists
+	needsSync := c.podExpectations.SatisfiedExpectations(key)
```

Comment:
Just for confirmation: Can you explain this?

Reply:
We return when it's false just above